### PR TITLE
Papia - Added approve button functionality on taskSuggestionPage

### PR DESCRIPTION
**Pull request by papiasharmin**
# Description
(PRIORITY HIGH) Yiyun/Jae: Create the ability for admin/owner users to approve and apply “suggested” changes on a task.
This was started here: [PR #721](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/pull/721)
Managers and Mentor classes can suggest changes on a task. These suggestions should create a notification (see below) for all Admins/Owners letting them know they need to be reviewed and approved. Once approved, a red bell should show up for anyone assigned the task AND for the person who made the original edits/suggestions
Any class that is other than a Volunteer AND on the same team should see the bell.

## Main changes explained:
moved fetchTaskEditSuggestion to Header component, to fetch latest notification on admin user log in, added approve button functionality.

## How to test:
1. check into current branch
2. do npm install  and npm run start:local to run this PR locally
3. log in as  manager , make a suggestion on any team member task, by clicking the task name on task tab.
4. log in as admin, see the notification count on header. click on it to go to the taskEditSuggeston page.
5. click on view suggestion button, you should see the suggestion you made as manager. 
6. on approving, will see a red bell for all member assigned with the task. and the suggestion will be deleted from suggestion list.

## Screenshots or videos of changes:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/86232710/793238c9-20ea-46b8-9209-d6a6198b1e6d

### CODE REVIEW

**Review by KurtisIvey — APPROVED** [review]
Works as intended! Good job

notification icon shows:
pr949 notification icon present

approval button works and presen:
pr949 button shows

**Comment by Jiangwei-shi** [comment]
First I log in to my manager account and suggest a task.
image
then I log in to my Adm account and see a request for the task
image
it shows my editor's information in green color
image
Finally, I click Approve button and then log in to my manager account. It successfully changes the information
image

However, during my test, I didn't find a way to assign a task to my manager account. I just share an existing task with my manager's account. Is easy to assign a project but not a task.

**Review by cvtqx — APPROVED** [review]
Functionality works as described.

Video:
https://www.dropbox.com/home/Highest%20Good%20Network%20Team/Olga/Videos?preview=PR%23949.mov

**Comment by beblicarl** [comment]
These PR changes work as described. Great work!

[screencast-localhost_3000-2023.06.28-19_45_48.webm](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/68056104/64a74402-1b64-466e-b4f7-56cf1338beb1)

**Review by beblicarl — APPROVED** [review]

**Comment by AuroraLovesWhales** [comment]
Hi @papiasharmin. I have tested your PR and everything works as intended except for one confusion I have. On the task description it says "Once approved, a red bell should show up for anyone assigned the task AND for the person who made the original edits/suggestions". So, I edited a task as a manager (the task does not include the manager itself as a resource), and approved the edit as an admin. But when I re log into to the manager account, I do not see the red bell because the manager is not assigned to the task. It only works when the manger is included in the task resource.

**Comment by papiasharmin** [comment]
> Hi @papiasharmin. I have tested your PR and everything works as intended except for one confusion I have. On the task description it says "Once approved, a red bell should show up for anyone assigned the task AND for the person who made the original edits/suggestions". So, I edited a task as a manager (the task does not include the manager itself as a resource), and approved the edit as an admin. But when I re log into to the manager account, I do not see the red bell because the manager is not assigned to the task. It only works when the manger is included in the task resource.

Hello, thank you for the review. A manager making suggestions on their team member's task, so when the red bell will appear on a team members task manager can see. I also got confused by that part.

**Comment by one-community** [comment]
> A manager making suggestions on their team member's task, so when the red bell will appear on a team members task manager can see. I also got confused by that part.

This is correct

File: src/components/TaskEditSuggestions/Components/TaskEditSuggestionRow.jsx [hunk 1]
```diff
@@ -11,11 +11,18 @@ export const TaskEditSuggestionRow = ({
...
-        {
-          <Link to={`/wbs/tasks/${taskEditSuggestion.oldTask._id}`}>
-            {taskEditSuggestion.oldTask.taskName}
-          </Link>
-        }
+        {taskEditSuggestion.oldTask.taskName}
+      </td>
+      <td>
+        <button onClick={handleToggleTaskEditSuggestionModal} 
+                style={{
```
Comment (cone-7) [inline src/components/TaskEditSuggestions/Components/TaskEditSuggestionRow.jsx:18]: Could you add a class?

**Review by tdkent — APPROVED** [review]
Local testing confirms that admin users can view and apply a **Suggest** submission made by a Manager on a given **Task**.

- Approval of the Suggest adds a notification to the admin user's dashboard. Clicking the notification prompts a popup showing the changes to the task.
- Other users assigned to the task will see a bell notification icon in their tasks panel. Clicking the icon prompts a popup showing the changes. Clicking the **Mark As Read** button clears the icon.
- Unclear from the description which users can and cannot see the bell notification, however from testing it appears it is visible to all team members and invisible to all others.

Testing Demo:

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/98129080/1ff105ba-dc64-4b16-a6e2-50275dfaac61

**Review by yaow62023 — APPROVED** [review]
The feature works as described see the video below, great work!
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/137429365/70300460-15ad-4271-9458-0221242b365a

**Comment by xaanders** [comment]
I have tested this PR.

- making suggestion as a manager to a task
- getting a notification as an admin
- seeing all the suggested changes
- aproving or rejecting the suggestion
- getting a red bell as manager or volunteer that disappears when marked as read

Manager can see bells of his teamates and if the teamate has read the notification, it disappears from managers list as well. 
Works as intended. Good job

[scrnli_07-07-2023_10-53-06.webm](https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/108314277/a5b2b90a-e784-4f2e-9626-81b4c8f45ecc)

**Review by xaanders — APPROVED** [review]

**Review by mahamakhan — APPROVED** [review]
Everything works as mentioned.

**Comment by one-community** [comment]
Thank you all, moved to final review!

**Review by shiva2096 — APPROVED** [review]
Changes working as expected.
Didn't saw the final comment so re-reviewed.

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/26687177/65ea245d-81a9-4505-8dc1-fb6b7b1ee616

**Comment by Lucas-E** [comment]
Great job everyone! Moving this one to final review!

**Review by AuroraLovesWhales — APPROVED** [review]
Hi @papiasharmin. I have reviewed this PR and everything is working as intended. Great job!

https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/44281916/89e8dda9-c374-4675-a1bb-458a8671da27

**Review by NidaZaki — APPROVED** [review]

**Pull request merged**